src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -170,7 +170,11 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
                 };
             }
 
-            $arguments[$i] = $payload;
+            if ($argument->metadata->isVariadic() && \is_array($payload)) {
```

Comment:
if the payload is not an array, we should throw a 422 instead, don't you think?
we should also use array_is_list so that we don't map to named arguments by mistake